File: C++/Step14-Math_3 Number Theory and Combinatorics/5086.cpp
//
//  5086.cpp
//  Step14-Math_3 Number Theory and Combinatorics
//
//  Created by 김예빈 on 2019. 9. 10..
//  Copyright © 2019년 김예빈. All rights reserved.
//

#include <iostream>
using namespace std;

int main(int argc, const char * argv[]) {
    cin.tie(NULL);
    ios::sync_with_stdio(false);
    
    int first, second;
    
    while(1) {
        cin >> first >> second;
        
        if(first == 0 && second == 0)
            break;
        
        if(first < second && second % first == 0)
            cout << "factor\n";
        else if(first > second && first % second == 0)
            cout << "multiple\n";
        else
            cout << "neither\n";
    }
    
    return 0;
}

File: BaekJoon Online Judge/Swift/Step09-Divisor, Multiple, Prime Number_1/5086.swift
//
//  5086.swift
//  Step16-Math_3 Number Theory and Combinatorics
//
//  Created by Yebin Kim on 2019/12/22.
//  Copyright © 2019 김예빈. All rights reserved.
//

import Foundation

while true {
    let input = readLine()!.split(separator: " ").map{ Int($0)! }.filter{ $0 > 0 }

    if input.count == 0 { break }
    
    let a = input[0]
    let b = input[1]
    
    if b % a == 0 {
        print("factor")
    } else if a % b == 0 {
        print("multiple")
    } else {
        print("neither")
    }
}
